orc8r/gateway/python/magma/ctraced/trace_manager.py

```diff
@@ -0,0 +1,104 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import logging
+import os
+import pathlib
+import subprocess
+
+from subprocess import SubprocessError
+
+TRACE_FILE_NAME = "call_trace.pcap"
+
+class TraceManager:
+    """
+    TraceManager is a wrapper for tshark specifically for starting and stopping
+    call/interface/subscriber traces.
+
+    Only a single trace can be captured at a time.
+    """
+    def __init__(self, config):
+        self._is_active = False # is call trace being captured
+        self._proc = None
+        self._file_name = None
+        self._trace_directory = config.get("trace_directory",
+                                           "/var/opt/magma/trace")
+        self._trace_interface = config.get("trace_interface", "eth0")
+
+
+    def start_trace(self) -> bool:
+        """
+        Start a call trace.
+        Captures all packets across the eth0 interface.
+        TODO (andreilee): Make configurable for trace type
+        :return: True if successful
+        """
+        if self._is_active:
+            logging.error("Failed to start trace: Trace already active")
+            return False
+        command = ["tshark"]
+
+        # Specify southbound interfaces
+        command.extend(["-i", self._trace_interface])
```

Comment:
it may be good to consider moving the creation of the trace to the trace object or, maybe better, to some kind of trace CommandFactory object. This way you could generate different strings with the command you want.  The trace manager should only be worried of getting a commandString

This way you could be extended in the future with more commands.

Reply:
Okay, done. Arguments between `tshark` and `tcpdump` aren't exactly equivalent, so the commands are to be generated by separate factory objects.